python/ray/serve/version.py

```diff
@@ -20,9 +20,9 @@ def __init__(self, code_version: Optional[str], user_config: Optional[Any] = Non
 
         self.user_config = user_config
         # TODO(simon): make this xlang compatible
-        pickled_user_config = pickle.dumps(user_config)
-        self.user_config_hash = crc32(pickled_user_config)
-        self._hash = crc32(pickled_user_config + self.code_version.encode("utf-8"))
+        serialized_user_config = str.encode(json.dumps(user_config))
```

Comment:
in order to make this consistent I think we need to add `sort_keys=True`

Reply:
Good catch, I added it in.